### PR TITLE
core: use new skeleton for running commands

### PR DESCRIPTION
**Pull request by subhamkrai**
Using the new skeleton using #13 as discussed.
With this PR, all the(possibly)Ceph commands and operator restart
are enabled.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
CI is failing due to the directory change in 1.8

**Comment by subhamkrai** [comment]
> CI is failing due to the directory change in 1.8

green now

File: kubectl-rook-ceph.sh [hunk 1]
```diff
@@ -16,33 +16,269 @@
...
+# HELPER FUNCTIONS
+####################################################################################################
 
-Commands:
-  ceph - run any 'ceph' CLI command as given
-    args  - any 'ceph' CLI argument or flag
+function print_usage () {
+  echo ""
+  echo "DESCRIPTION"
+  echo "  blah blah blah"
```
Comment (travisn) [inline kubectl-rook-ceph.sh:26]: Nice description :)

**Review by travisn — COMMENTED** [review]
Since i'm out this week, not marking as changes requested...

File: kubectl-rook-ceph.sh [hunk 2]
```diff
@@ -16,33 +16,269 @@
...
+  echo "  kubectl rook-ceph <main args> <command> <command args>"
+  echo "MAIN ARGS"
+  echo "  -v, --verbose               : output verbose info"
+  echo "  -h, --help                  : output help text"
+  echo "  -n, --namespace='rook-ceph' : the namespace of the CephCluster"
+  echo "COMMANDS"
+  echo "  help           : output help text"
+  echo "  ceph <args>    : call a 'ceph' CLI command with arbitrary args"
+  echo "  status         : show status of the CephCluster and its resources"
+  echo "    -l, --long     : show long status"
```
Comment (leseb) [inline kubectl-rook-ceph.sh:37]: wrong indent?

File: kubectl-rook-ceph.sh [hunk 3]
```diff
@@ -16,33 +16,269 @@
...
+  echo "  operator <subcommand>..."
+  echo "    restart      : restart the Rook-Ceph operator"
```
Comment (leseb) [inline kubectl-rook-ceph.sh:39]: wrong indent?

Comment (subhamkrai) [reply kubectl-rook-ceph.sh:39]: this is a subcommand of `operator` that's why it is inside the operator.

File: kubectl-rook-ceph.sh (hunk 2 above)
Comment (subhamkrai) [reply kubectl-rook-ceph.sh:37]: this is `flag of status` that's why it is inside status.

File: kubectl-rook-ceph.sh [hunk 4]
```diff
@@ -16,33 +16,269 @@
...
+  print_usage
+}
+
+####################################################################################################
+# 'kubectl rook-ceph ceph ...' command
+####################################################################################################
+
+function run_ceph_command () {
+  # do not call end_of_command_parsing here because all remaining input is passed directly to 'ceph'
+  kubectl --namespace "$NAMESPACE" exec deploy/rook-ceph-operator -- /bin/bash -c " ceph ${*} --conf=/var/lib/rook/rook-ceph/rook-ceph.config"
```
Comment (BlaineEXE) [inline kubectl-rook-ceph.sh:159]: Remind me why this is `/bin/bash -c "ceph $* --conf=...` instead of `ceph --conf=... "$@"` (why /bin/bash, and why --conf arg after the rest of the command?

Comment (subhamkrai) [reply kubectl-rook-ceph.sh:159]: `--conf` is to point to the ceph config file which is required to run ceph commands and I think when we run any command inside pod we pass `bash/sh -c`(`-c` for the command).

Comment (BlaineEXE) [reply kubectl-rook-ceph.sh:159]: Let me rephrase:
1. Why is `--conf` the last argument and not the first?
2. Why is `/bin/bash -c` necessary at all? This works fine for me and is less complex: `kubectl exec deploy/rook-ceph-operator -- ceph --conf=... <command>`

File: kubectl-rook-ceph.sh [hunk 5]
```diff
@@ -16,33 +16,269 @@
...
+      ;;
+    "-h"|"--help")
+      flag_no_value "$flag" "$val"
+      run_help_command
+      exit 0 # unique for the help flag; stop parsing everything and exit with success
+      ;;
+    "-v"|"--verbose")
+      # just an example
+      flag_no_value "$flag" "$val"
+      echo "VERBOSE MODE"
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply kubectl-rook-ceph.sh]: for now, I have removed `verbose` I'll enable once it is required

File: kubectl-rook-ceph.sh [hunk 6]
```diff
@@ -16,33 +16,269 @@
...
+      ;;
+    ceph)
+      run_ceph_command "$@"
+      ;;
+    operator)
+      run_operator_command "$@"
+      ;;
+    status)
+      run_status_command "$@"
+      ;;
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply kubectl-rook-ceph.sh]: disabled

File: kubectl-rook-ceph.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (subhamkrai) [reply kubectl-rook-ceph.sh]: removed. will add when `status` command is added

File: kubectl-rook-ceph.sh [hunk 7]
```diff
@@ -16,33 +16,269 @@
...
+    *)
+      fail_error "'operator' subcommand '$subcommand' does not exist"
+      ;;
+  esac
+}
+
+function run_operator_restart_command () {
+  end_of_command_parsing "$@" # end of command tree
+  kubectl --namespace "$NAMESPACE" rollout restart deploy/rook-ceph-operator
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply kubectl-rook-ceph.sh:179]: for now, I'm removing it as when operator restart is run it will show relevant log by kubectl so not required IMO for now. something like `deployment.apps/rook-ceph-operator restarted`

File: kubectl-rook-ceph.sh (hunk 4 above)
Comment (subhamkrai) [reply kubectl-rook-ceph.sh]: > Let me rephrase:
> 
> 1. Why is `--conf` the last argument and not the first?
I liked in this way `ceph <some command> --conf=....`
> 2. Why is `/bin/bash -c` necessary at all? This works fine for me and is less complex: `kubectl exec deploy/rook-ceph-operator -- ceph --conf=... <command>`
right, I have removed `/bin/bash -c`
